src/pipe_log.rs

```
@@ -349,175 +369,187 @@ impl FilePipeLog {
-    }
-
-    fn total_size(&self) -> usize {
-        let manager = self.appender.rl();
-        (manager.active_file_num - manager.first_file_num) as usize * self.rotate_size
-            + manager.active_log_size
-    }
-
-    // For recovery.
-    fn read_next_file_impl(&mut self, name_suffix: &'static str) -> Result<Option<Vec<u8>>> {
-        let manager = match name_suffix {
-            LOG_SUFFIX => self.appender.rl(),
-            REWRITE_SUFFIX => self.rewriter.rl(),
-            _ => unreachable!(),
-        };
-
-        if self.current_read_file_num == 0 {
-            self.current_read_file_num = manager.first_file_num;
-        }
-        if self.current_read_file_num > manager.active_file_num {
-            return Ok(None);
-        }
-
+    fn read_whole_file(&self, queue: LogQueue, file_num: u64) -> Result<Vec<u8>> {
         let mut path = PathBuf::from(&self.dir);
-        path.push(generate_file_name(self.current_read_file_num, name_suffix));
+        path.push(generate_file_name(file_num, self.get_name_suffix(queue)));
         let meta = fs::metadata(&path)?;
         let mut vec = Vec::with_capacity(meta.len() as usize);
 
         // Read the whole file.
         let mut file = File::open(&path)?;
         file.read_to_end(&mut vec)?;
+        Ok(vec)
+    }
+
+    fn purge_to(&self, queue: LogQueue, file_num: u64) -> Result<usize> {
+        let (mut manager, name_suffix) = match queue {
+            LogQueue::Append => (self.appender.wl(), LOG_SUFFIX),
+            LogQueue::Rewrite => (self.rewriter.wl(), REWRITE_SUFFIX),
+        };
+        let purge_count = manager.purge_to(file_num)?;
+        drop(manager);
+
+        for file_num in (file_num - purge_count as u64)..file_num {
+            let mut path = PathBuf::from(&self.dir);
+            path.push(generate_file_name(file_num, name_suffix));
+            if let Err(e) = fs::remove_file(&path) {
+                warn!("Remove purged log file {:?} fail: {}", path, e);
```

Is this safe swallowing the error here?